geoproxy/main.go

```diff
@@ -5,19 +5,44 @@ package main
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"log"
+	"math"
 	"net/http"
 	"net/http/httputil"
 	"net/url"
 	"os"
-	"sort"
 	"strconv"
-	"fmt"
+	"strings"
 )
 
-var data = map[int]map[int]string{}
-var sorted_lat []int
+// Coord struct represents the lat-lng coordinate
+type Coord struct {
+	Lat, Lng float64
+}
+
+// Key struct is for map pointing to correct server
+type CoordRange struct {
+	Low, High float64
+}
+
+// View request passes top-left and bottom-right coords
+// "latlng1": [20, 0],
+// "latlng2": [0, 80]
+type viewRequestPayloadStruct struct {
+	LatLng1 string `json:"latlng1"`
+	LatLng2 string `json:"latlng2"`
+}
+
+// Submit request passes the coord to post
+// "coordinate": [-20, -20]
+type submitRequestPayloadStruct struct {
```

Comment:
There are apparently a couple other things that the client is sending in the request:
[https://blueteam-comp413.slack.com/archives/CN2QQFVHQ/p1570810596000900](url)
Should we add these?

Reply:
Sorry, link isn't working. However, I did test it by adding a test parameter in the curl request, and the message still sends correctly. So I think we should only decode parameters we need to route with.

Reply:
I imagine that if it isn't important for routing we shouldn't  include those elements? I guess the question is whether we want to validate requests at this point or if we're assuming that all incoming requests are correct. I feel like we should stick to the latter for now, but theoretically we wouldn't want to forward incorrect/invalid requests to the server?

Reply:
My bad, Kevin posted the following in the backend channel:
coordinate: Coordinate (required)
timestamp: datetime (required, in UTC)
text: string (optional)
Image: image (optional)

But yea what Shashank says makes sense